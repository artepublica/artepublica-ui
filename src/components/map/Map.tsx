import { FontAwesome } from '@expo/vector-icons';
import { useBottomTabBarHeight } from '@react-navigation/bottom-tabs';
import { useHeaderHeight } from '@react-navigation/elements';
import { useWindowDimensions } from 'react-native';
import MapView, { Marker } from 'react-native-maps';

type MapWrapperProps = {
    markers?: { position: { latitude: string; longitude: string } }[];
};

function Map({ markers }: MapWrapperProps): JSX.Element {
    const { height, width } = useWindowDimensions();
    const bottomTabBarHeight = useBottomTabBarHeight();
    const headerHeight = useHeaderHeight();
    const center = { latitude: -22.910972, longitude: -43.17156, latitudeDelta: 0.5, longitudeDelta: 0.5 };

    return (
        <MapView
            zoomControlEnabled
            zoomEnabled
            zoomTapEnabled
            initialRegion={center}
            testID="mapView"
            style={{ height: height - headerHeight - bottomTabBarHeight, width }}
        >
            {markers?.map((marker, index) => (
                <Marker key={index} coordinate={{ latitude: parseFloat(marker.position.latitude), longitude: parseFloat(marker.position.longitude) }}>
                    <FontAwesome name="map-marker" size={24} color="black" />
                </Marker>
            ))}
        </MapView>
    );
}

export default Map;
