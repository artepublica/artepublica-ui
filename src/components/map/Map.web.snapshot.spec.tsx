import { create } from 'react-test-renderer';

import Map from './Map.web';

describe('<Map.web />', () => {
    it('default snapshot', () => {
        const json = create(<Map />).toJSON();

        expect(json).toMatchSnapshot();
    });
});
