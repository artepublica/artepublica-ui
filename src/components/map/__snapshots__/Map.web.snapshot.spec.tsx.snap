// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Map.web /> default snapshot 1`] = `null`;
